assets/blocks/course-categories-block/course-categories-edit.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { unescape } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useMemo } from 'react';
+import useColors from './hooks/use-colors';
+import useCourseCategories from './hooks/use-course-categories';
+import {
+	withColorSettings,
+	withDefaultColor,
+} from '../../shared/blocks/settings';
+import { compose } from '@wordpress/compose';
+
+export function CourseCategoryEdit( props ) {
+	const { context, attributes } = props;
+	const { textAlign } = attributes;
+	const { postId } = context;
+	const term = 'course-category';
+
+	const {
+		postTerms: categories,
+		hasPostTerms: hasCategories,
+		isLoading,
+	} = useCourseCategories( postId );
+
+	const { textColor, backgroundColor } = useColors( props );
```

Comment:
Do we need the `useColors` hook? I think `withColorSettings` takes care of it. Can you kindly check? I tried it locally and it worked for me. use can change this line to `const { textColor, backgroundColor } = props;`, then in the `block.json` file add the following lines under `attributes` object definition, 
```
"textColor": {
            "type": "string"
  },
  "backgroundColor": {
      "type": "string"
  },
```
For me it worked and I could see the color getting saved and updated properly in the editor. To render it in the frontend properly, you'll probably also need to change in the PHP side's render function for this block as you're currently using `$attributes['categoryStyle']` for rendering the styling, you'll find the styles directly under `$attributes` then so you won't probably need `'categoryStyle'`. I'm not sure about the whole thing, just thought if we reuse the current hooks and components without rewriting, that'll be helpful for us, can you kindly check?

Reply:
@Imran92 The problem happens when we set a custom color to an element using the color picker.
The current `withColorSettings` is only storing the already defined colors, for custom colors, it is expecting that the style should be added directly on the block root element:

<img width="933" alt="Screen Shot 2022-08-18 at 18 43 22" src="https://user-images.githubusercontent.com/38718/185500458-dec5bbf2-ebbf-4d77-830f-bf09e5cca625.png">

The categories block case is different because the root element is not the element that should contain the custom color.

Both  `Sensei_Block_Helpers::build_styles()` and `Sensei_Block_Helpers::render_style_attributes()` are also assuming it

Reply:
This [code](https://github.com/Automattic/sensei/blob/a7ee53b0da7c809c32b94f3e4d11e84ec58789dd/includes/blocks/class-sensei-block-helpers.php#L52) apparently supports a custom field, but it only helps one field and we need to have two.

Reply:
I've gone through the code a bit, I think you can add two more properties in `block.json` inside the `attributes` definition and it'll work for custom colors as well, they are as follows
```
"customTextColor": {
    "type": "string"
},
"customBackgroundColor": {
    "type": "string"
},
```
Can you kindly check? Thanks :)

Reply:
> <img alt="Screen Shot 2022-08-18 at 18 43 22" width="933" src="https://user-images.githubusercontent.com/38718/185500458-dec5bbf2-ebbf-4d77-830f-bf09e5cca625.png">
> 
> The categories block case is different because the root element is not the element that should contain the custom color.

I'm not sure if I'm right, but I think the attributes belong to the block and not any individual root or other HTML element inside it. It's up to us what we decide to do with the attribute and on which element we want to apply it. If a block is an accumulation of multiple blocks, like our `Course Actions` block which has multiple inner blocks like `Take Course`, `Countinue` etc, then we can have one style for the root or parent block and other styles for the individual inner blocks, but that's because the inner items are `blocks`. I'm not entirely sure, but that's what I understand ☺️

Reply:
@Imran92  I have the same discussion with @donnapep
The main point is that we won't have inner blocks, because, for example, we wont to have the ability to set the background color of each category. 

What we want is just to have the ability to set certain attributes to all categories, the complexity to support inner blocks vs the value delivered doesn't seem good.

Reply:
> @Imran92 I have the same discussion with @donnapep The main point is that we won't have inner blocks, because, for example, we wont to have the ability to set the background color of each category.
> 
> What we want is just to have the ability to set certain attributes to all categories, the complexity to support inner blocks vs the value delivered doesn't seem good.

Exactly 👍  I think we are on the same page here that we don't want individual colors for each category, all categories will have same color. My point is I think we can achieve it using existing components and functions as mentioned [here](https://github.com/Automattic/sensei/pull/5455#discussion_r949671586) and [here](https://github.com/Automattic/sensei/pull/5455#discussion_r949299773) for both predefined and custom colors, which will be applicable for all the categories inside

Reply:
Ok, let's do a pair session because I suspect we are talking about the same thing in different ways.